File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/TestToolDetails.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;


/**
 * Base model for testing any tool.  {@code toolType} drives the subtype.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.annotation.JsonTypeInfo(use=com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, include=com.fasterxml.jackson.annotation.JsonTypeInfo.As.PROPERTY, property="toolType", defaultImpl=TestToolDetails.class)
@com.fasterxml.jackson.annotation.JsonSubTypes({
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestCustomToolDetails.class, name = "CUSTOM"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestHttpToolDetails.class, name = "HTTP"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestSqlToolDetails.class, name = "SQL"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestRagToolDetails.class, name = "RAG"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestMcpToolDetails.class, name = "MCP"),
    @com.fasterxml.jackson.annotation.JsonSubTypes.Type(value = TestPromptToolDetails.class, name = "PROMPT")
})

public class TestToolDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"agentFlowId"})
    protected TestToolDetails(String agentFlowId) {
        super();
        this.agentFlowId = agentFlowId;
    }




        /**
     * Agent flow id for which the tool is being tested
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowId")
    private final String agentFlowId;

        /**
     * Agent flow id for which the tool is being tested
     * @return the value
     **/
    
    public String getAgentFlowId() {
        return agentFlowId;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("TestToolDetails(");
        sb.append("agentFlowId=").append(String.valueOf(this.agentFlowId));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof TestToolDetails)) {
            return false;
        }

        TestToolDetails other = (TestToolDetails) o;
        return java.util.Objects.equals(this.agentFlowId, other.agentFlowId);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.agentFlowId == null ? 43 : this.agentFlowId.hashCode());
        return result;
    }

    /**
     * Which kind of tool this test is for (chooses the subtype).
     **/
    public enum ToolType {
        Sql("SQL"),
        Prompt("PROMPT"),
        Rag("RAG"),
        Mcp("MCP"),
        Http("HTTP"),
        Custom("CUSTOM"),
        ;

        

        private final String value;
        private static java.util.Map<String, ToolType> map;

        static {
            map = new java.util.HashMap<>();
            for (ToolType v : ToolType.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        ToolType(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static ToolType create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid ToolType: " + key);
        }
    };
}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/test_tool_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class TestToolDetails(object):
    """
    Base model for testing any tool.  `toolType` drives the subtype.
    """

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "SQL"
    TOOL_TYPE_SQL = "SQL"

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "PROMPT"
    TOOL_TYPE_PROMPT = "PROMPT"

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "RAG"
    TOOL_TYPE_RAG = "RAG"

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "MCP"
    TOOL_TYPE_MCP = "MCP"

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "HTTP"
    TOOL_TYPE_HTTP = "HTTP"

    #: A constant which can be used with the tool_type property of a TestToolDetails.
    #: This constant has a value of "CUSTOM"
    TOOL_TYPE_CUSTOM = "CUSTOM"

    def __init__(self, **kwargs):
        """
        Initializes a new TestToolDetails object with values from keyword arguments. This class has the following subclasses and if you are using this class as input
        to a service operations then you should favor using a subclass over the base class:

        * :class:`~oci.aidataplatform_dp.models.TestCustomToolDetails`
        * :class:`~oci.aidataplatform_dp.models.TestHttpToolDetails`
        * :class:`~oci.aidataplatform_dp.models.TestSqlToolDetails`
        * :class:`~oci.aidataplatform_dp.models.TestRagToolDetails`
        * :class:`~oci.aidataplatform_dp.models.TestMcpToolDetails`
        * :class:`~oci.aidataplatform_dp.models.TestPromptToolDetails`

        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param agent_flow_id:
            The value to assign to the agent_flow_id property of this TestToolDetails.
        :type agent_flow_id: str

        :param tool_type:
            The value to assign to the tool_type property of this TestToolDetails.
            Allowed values for this property are: "SQL", "PROMPT", "RAG", "MCP", "HTTP", "CUSTOM"
        :type tool_type: str

        """
        self.swagger_types = {
            'agent_flow_id': 'str',
            'tool_type': 'str'
        }

        self.attribute_map = {
            'agent_flow_id': 'agentFlowId',
            'tool_type': 'toolType'
        }

        self._agent_flow_id = None
        self._tool_type = None

    @staticmethod
    def get_subtype(object_dictionary):
        """
        Given the hash representation of a subtype of this class,
        use the info in the hash to return the class of the subtype.
        """
        type = object_dictionary['toolType']

        if type == 'CUSTOM':
            return 'TestCustomToolDetails'

        if type == 'HTTP':
            return 'TestHttpToolDetails'

        if type == 'SQL':
            return 'TestSqlToolDetails'

        if type == 'RAG':
            return 'TestRagToolDetails'

        if type == 'MCP':
            return 'TestMcpToolDetails'

        if type == 'PROMPT':
            return 'TestPromptToolDetails'
        else:
            return 'TestToolDetails'

    @property
    def agent_flow_id(self):
        """
        **[Required]** Gets the agent_flow_id of this TestToolDetails.
        Agent flow id for which the tool is being tested


        :return: The agent_flow_id of this TestToolDetails.
        :rtype: str
        """
        return self._agent_flow_id

    @agent_flow_id.setter
    def agent_flow_id(self, agent_flow_id):
        """
        Sets the agent_flow_id of this TestToolDetails.
        Agent flow id for which the tool is being tested


        :param agent_flow_id: The agent_flow_id of this TestToolDetails.
        :type: str
        """
        self._agent_flow_id = agent_flow_id

    @property
    def tool_type(self):
        """
        **[Required]** Gets the tool_type of this TestToolDetails.
        Which kind of tool this test is for (chooses the subtype).

        Allowed values for this property are: "SQL", "PROMPT", "RAG", "MCP", "HTTP", "CUSTOM"


        :return: The tool_type of this TestToolDetails.
        :rtype: str
        """
        return self._tool_type

    @tool_type.setter
    def tool_type(self, tool_type):
        """
        Sets the tool_type of this TestToolDetails.
        Which kind of tool this test is for (chooses the subtype).


        :param tool_type: The tool_type of this TestToolDetails.
        :type: str
        """
        allowed_values = ["SQL", "PROMPT", "RAG", "MCP", "HTTP", "CUSTOM"]
        if not value_allowed_none_or_none_sentinel(tool_type, allowed_values):
            raise ValueError(
                "Invalid value for `tool_type`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._tool_type = tool_type

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
